### PR TITLE
feat: migration from `jsonrpc-v2` to `actix`

### PR DESCRIPTION
**Pull request by frolvanya**
This migration would help us to:
- Implement custom status codes (200, 400, 404, 408 and 500) like we do it in jsonrpc
- Build a unified server configuration across jsonrpc and read-rpc
- Implement a public jsonrpc trait to have same definition of rpc methods

### CODE REVIEW

File: rpc-server/src/main.rs
Comment (frolvanya) [inline rpc-server/src/main.rs:368]: @khorolets Any ideas how to insert all methods now? I don't want to hardcode them here

File: rpc-server/src/main.rs
Comment (frolvanya) [inline rpc-server/src/main.rs:76]: `RpcViewStatePaginatedRequest` doesn't support `RpcRequest` trait and it cannot be parsed. Also we can't simply implement this trait now, because it requires `Params` structure to be accessible from read-rpc. Unfortunately, for now this is not the case as `Params` structure is declared as `pub(crate)`, so we can't use it for now. So we either need to wait until my PR related to splitting jsonrpc into public trait will be merged (because I've moved `Params` structure there and it will be public to everybody) or think about another solution

**Comment by frolvanya** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@frolvanya Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Congratulatory messages are spent out! Your results have exceeded our imagination!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/frolvanya)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/frolvanya)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by khorolets — APPROVED** [review]

**Comment by khorolets** [comment]
@race-of-sloths score 8

**Pull request merged**
